worldedit-core/src/main/java/com/sk89q/worldedit/function/block/SnowSimulator.java

```diff
@@ -116,6 +116,9 @@ public boolean apply(BlockVector3 position, int depth) throws WorldEditException
             }
             return false;
         }
+        if (this.extent instanceof World && !((World) this.extent).canPlaceAt(abovePosition, snow)) {
```

Comment:
Just take and store `World`, if we want to use `Extent` in the future we can just change the constructor (that's backwards-compatible).

Reply:
Due to the extent here being the one that applies blocks, we'd need to pass in a second world to do that. Instead, I've made this require a world or EditSession to be passed in